desk/app/reel.hoon

```diff
@@ -221,7 +301,126 @@
   |=  [=wire =sign:agent:gall]
   ^-  (quip card _this)
   =/  =(pole knot)  wire
-  ?+  pole  (on-agent:def wire sign)
+  ?+    pole  (on-agent:def wire sign)
+      [%update %contact ~]
+    ?>  ?=(%poke-ack -.sign)
+    ?~  p.sign  `this
+    %-  (fail:log %poke-ack 'profile update failed' u.p.sign)
+    `this
+  ::
+      [%update %group ~]
+    ?>  ?=(%poke-ack -.sign)
+    ?~  p.sign  `this
+    %-  (fail:log %poke-ack 'group meta update failed' u.p.sign)
+    `this
+  ::
+      [%contacts ~]
+    ?+    -.sign  (on-agent:def wire sign)
+        %kick
+      =^  caz=(list card)  subs
+        (~(subscribe s [subs bowl]) /contacts [our.bowl %contacts] contacts-path &)
+      [caz this]
+    ::
+        %watch-ack
+      ?~  p.sign  `this
+      %-  (fail:log %watch-ack 'failed to subscribe to contacts' u.p.sign)
+      `this
+    ::
+        %fact
+      =+  !<(=response:t q.cage.sign)
+      ?.  ?=(%self -.response)  `this
+      =*  profile  con.response
+      =>
+        |%
+        ::  +hand: choose right and test equality
+        ++  hand
+          |*  [a=(unit) b=(unit)]
+          ^-  [_b ?]
+          ?~  a  [b ?=(~ b)]
+          ?~  b  [~ ?=(~ a)]
+          [b =(u.a u.b)]
+        --
+      ::  check profile for relevant changes
+      ::
+      =/  [nickname=(unit @t) axe=?]
+        %+  hand
+          (~(get cy:t our-profile) %nickname %text)
+        (~(get cy:t profile) %nickname %text)
+      =/  [avatar=(unit @t) bax=?]
+        %+  hand
+          (~(get cy:t our-profile) %avatar %look)
+        (~(get cy:t profile) %avatar %look)
+      =/  [color=(unit @ux) cax=?]
+        %+  hand
+          (~(get cy:t our-profile) %color %tint)
+        (~(get cy:t profile) %color %tint)
+      =.  our-profile  profile
+      ::  nothing relevant has changed, skip the update
+      ?:  &(axe bax cax)  `this
```

Comment:
We poke the bait service conditionally here, but not for group metadata changes, because our profile may contain a bunch of irrelevant/not-displayed info, but for groups all the metadata gets used always?